### ISSUE TITLE
Check Current App

### ISSUE THREAD

**Issue by MelbourneDeveloper** [LedgerHQ/ledger-dotnet-api#19]
Is there a way to check what app is running on the device? More specifically which coin? 

If not, is this a limitation of the firmware? Or is this just not implemented in C#?

**Comment by NicolasDorier** [comment]
That is a good question. Have you tried the `GetFirmewareInfo` ?

**Comment by MelbourneDeveloper** [comment]
Thanks. I can see that messages are defined in LedgerWalletConstants. I don't think GetCoinVersion is defined there. Where can I see the full list of messages?

I found this file:
https://github.com/LedgerHQ/ledger-javacard/blob/master/src/com/ledger/wallet/LedgerWalletApplet.java

But, this also does not have GetCoinVersion .

**Comment by NicolasDorier** [comment]
just define it

**Comment by MelbourneDeveloper** [comment]
Ah yes! I see what you mean. Actually the file can be found here:

https://github.com/LedgerHQ/blue-app-btc/blob/master/include/btchip_apdu_constants.h

**Comment by MelbourneDeveloper** [comment]
@NicolasDorier , isn't this call (BTCHIP_INS_GET_COIN_VER) an APDU which is specific to the Bitcoin app?

Wouldn't I already have to be using the Bitcoin app to call this?

https://github.com/LedgerHQ/blue-app-btc

**Comment by NicolasDorier** [comment]
Yeah, take example on the `GetFirmwareVersion` which already exist in the `LedgerClient`. This is not that complicated.

**Comment by MelbourneDeveloper** [comment]
Yes, actually, I uninstalled and reinstalled the Bitcoin app today. 

I'm thinking that this is not possible because even the Ledger Live app doesn't do this. If I want to add a coin to my portfolio, I have to select it from a drop down list. It can't just automatically detect what app I am using.

**Comment by NicolasDorier** [comment]
Strange thing is that the code for it exists: https://github.com/LedgerHQ/blue-app-btc/blob/master/src/btchip_apdu_get_coin_version.c

**Comment by MelbourneDeveloper** [comment]
That code was committed Nov 19, 2016 so I doubt it's a problem with the app being out of date...

**Comment by MelbourneDeveloper** [comment]
Is it worth logging an issue with the repo?

**Comment by NicolasDorier** [comment]
I tested in my repo, your code is working on my ledger.

**Comment by MelbourneDeveloper** [comment]
Is that nano? Firmware 1.4.2?

**Comment by NicolasDorier** [comment]
App version 1.2.6, firmware 1.4.2

**Comment by MelbourneDeveloper** [comment]
Mine is 1.2.8. It might be broken in this version...

**Comment by MelbourneDeveloper** [comment]
Hmm... It's working now.

I uninstalled and reinstalled the app and now it's OK...

OK. Pull request will be on its way soon I think

**Comment by NicolasDorier** [comment]
nice!

**Comment by MelbourneDeveloper** [comment]
It works fine with Bitcoin and Bitcoin Gold but it doesn't seem to work with any of the other apps. I'm guessing that this APDU is an optional message. I'm thinking I'd better log issues with each of the other apps like Ethereum and so on?

Is there a list of all the app repos?

**Comment by NicolasDorier** [comment]
All bitcoin clones are on  https://github.com/LedgerHQ/blue-app-btc , eth you need to look on somewhere in https://github.com/LedgerHQ/

**Comment by NicolasDorier** [comment]
I expect it to work on all bitcoin clones.

**Comment by MelbourneDeveloper** [comment]
Cool. Maybe I have to update all the apps...

This is probably the ETH app right?
https://github.com/LedgerHQ/blue-app-eth

**Comment by NicolasDorier** [comment]
yes

**Comment by MelbourneDeveloper** [comment]
A quick look tells me that it is not implemented there but I could be wrong.

It seems like this APDU was only half implemented. If LedgerHQ wants to implement this, I think they'd need to force all the apps to implement it.

**Comment by MelbourneDeveloper** [comment]
This is really a wider issue though...

Come to think of it, even GetWalletPubKey doesn't work for most apps. Even Bitcoin Gold expects a different parameter and throws an exception. It doesn't work with Ethereum etc.

There seems to be no real standardized interface across apps.

**Comment by NicolasDorier** [comment]
I am surprised Bitcoin Gold expects a different parameter, what parameter?

**Comment by NicolasDorier** [comment]
btw, I pushed new package including your changes. (2.0.0.1)
